### PR TITLE
Remove build dependencies. Fixes #985 and #958

### PR DESCRIPTION
**Pull request by zerolab**
This PR builds on top of #985. It adds a "Dependecies" section to the "Installation" page, making it explcit that django-libsass and Pillow must be added to requirements.

### CODE REVIEW

File: docs/getting_started/installation.rst [hunk 1]
```diff
@@ -6,10 +6,50 @@ Installation
...
+Dependencies
+============
+
+The basic Wagtail installation is pure Python. No build tools are required on the host machine.
+If you are installing Wagtail differently (e.g. from the Git repo), make you meet the following dependencies:
+
+Required
+--------
+
+ * Django>=1.7.0,<1.8
```
Comment (kaedroho) [inline docs/getting_started/installation.rst]: There's a good chance that we will forget to update this list...

Comment (zerolab) [inline docs/getting_started/installation.rst]: What is the best way to mention the dependecies, without forgetting to update the list? Should I link to https://github.com/torchbox/wagtail/blob/master/setup.py#L30 ?

Comment (kaedroho) [inline docs/getting_started/installation.rst]: When people are installing from git, they should run ``python setup.py install`` from the root of the repo which will also install all of Wagtails dependencies.

See: https://docs.djangoproject.com/en/1.7/topics/install/#installing-an-official-release-manually

Comment (zerolab) [inline docs/getting_started/installation.rst]: Thanks, @kaedroho. Changed with a link to setup.py for the full list of dependencies + added the note on running `python setup.py install``

**Comment by kaedroho** [comment]
Looks good to me!

**Comment by tomdyson** [comment]
Thanks, @zerolab!

**Pull request merged**